src/CreditCard.php

```diff
@@ -127,9 +127,9 @@ public static function validCreditCard($number, $type = null)
         }
 
         $ret['validation'] = array(
-            'pattern' => (bool)self::validPattern($number, $type),
-            'length' => self::validLength($number, $type),
-            'luhn' => self::validLuhn($number, $type),
+            'pattern' => !empty($type) && self::validPattern($number, $type),
+            'length' => !empty($type) && self::validLength($number, $type),
+            'luhn' => !empty($type) && self::validLuhn($number, $type),
```

Comment:
Wouldn't this be unnecessary due to lines 117-119? We're also losing the cast to bool from validPattern().